v0x04/match_fields.py

```diff
@@ -168,16 +252,23 @@ class MatchNwSrc(MatchField):
 
     def as_of_tlv(self):
         """Return a pyof OXM TLV instance."""
-        value_bytes = IPAddress(self.value).pack()
-        return OxmTLV(oxm_field=self.oxm_field, oxm_value=value_bytes)
+        ip_addr = IPAddress(self.value)
+        value_bytes = ip_addr.pack()
+        if ip_addr.netmask < 32:
+            value_bytes += mask_to_bytes(ip_addr.netmask, 32)
+        return OxmTLV(oxm_field=self.oxm_field,
+                      oxm_hasmask=ip_addr.netmask < 32,
+                      oxm_value=value_bytes)
 
     @classmethod
     def from_of_tlv(cls, tlv):
         """Return an instance from a pyof OXM TLV."""
         ip_address = IPAddress()
         ip_address.unpack(tlv.oxm_value)
-        ip_str = str(ip_address)
-        return cls(ip_str)
+        value = str(ip_address)
```

Comment:
In the mac address fields you first attributed the value of `str(hw_address)` to a `addr_str` variable, and then attributed it to `value`. Here, you are attributing `str(ip_address)` directly into `value`. You should do both attributions the same way.